src/aws_secret.cpp

```diff
@@ -73,7 +73,12 @@ class DuckDBCustomAWSCredentialsProviderChain : public Aws::Auth::AWSCredentials
 				if (!profile.empty()) {
 					AddProvider(std::make_shared<Aws::Auth::STSProfileCredentialsProvider>(profile));
 				} else if (!assume_role_arn.empty()) {
-					AddProvider(std::make_shared<Aws::Auth::STSAssumeRoleCredentialsProvider>(assume_role_arn, Aws::String(), Aws::String(), Aws::Auth::DEFAULT_CREDS_LOAD_FREQ_SECONDS, sts_client));
+					if (!external_id.empty()) {
+                        AddProvider(std::make_shared<Aws::Auth::STSAssumeRoleCredentialsProvider>(
```

Comment:
nit: some whitespace

Reply:
hey @Tmonster did you want me to add or remove whitespace? Not clear and the git diff makes it look funny.

I've merged line 77 and 78 so that its all on one line like other calls